Dockerfile

```diff
@@ -6,6 +6,12 @@
 
 ARG TOOLCHAIN
 
+FROM fedora:39 AS zfs
+COPY hack/zfs-generate.sh /
+RUN dnf install -y zfs-fuse && rm -rf /var/cache/dnf
```

Comment:
I feel that given the value available, can we simply put the result of this to `blkid/testdata/zfs.img`, and somewhere (like `blkid/testdata/README.md`) put an instruction how to reproduce the file using this small `Dockerfile`.

I think this would still keep it easily reproducible, but it won't make our automated `Dockerfile` generator more complicated

Reply:
we can add this in `.kres.yaml`

```bash
---
kind: auto.CustomSteps
spec:
  steps:
    - name: zfs-img
      toplevel: true
---
kind: custom.Step
name: zfs-img
spec:
  makefile:
    enabled: true
  docker:
    enabled: true
    stages:
      - name: zfs-img
        description: Creates the ZFS image
        from: fedora:39
        steps:
          - copy:
              src: hack/zfs-generate.sh
              dst: /
          - script:
              command: dnf install -y zfs-fuse && rm -rf /var/cache/dnf
          - script:
              command: --security=insecure /zfs-generate.sh
```

and run `make local-zfs-img` to generate, and add to readme, makes things easier

Reply:
also `lvm2` is missing in `.kres.yaml`

```yaml
---
kind: golang.Toolchain
spec:
    extraPackages:
        - cdrkit
        - cryptsetup
        - dosfstools
        - e2fsprogs
        - parted
        - util-linux
        - xfsprogs
        - lvm2

```

Reply:
okay, this is better:

```yaml
---
kind: auto.CustomSteps
spec:
  steps:
    - name: zfs-img
      toplevel: true
---
kind: custom.Step
name: zfs-img
spec:
  makefile:
    enabled: true
    script:
      - |
        @$(MAKE) local-$@ DEST=$(ARTIFACTS)  TARGET_ARGS="--allow security.insecure"
  docker:
    enabled: true
    stages:
      - name: zfs-img-gen
        description: Creates the ZFS image
        from: fedora:39
        steps:
          - script:
              command: dnf install -y zfs-fuse && rm -rf /var/cache/dnf
          - script:
              command: |
                --security=insecure zfs-fuse & \
                dd if=/dev/zero of=/tmp/zfs.img bs=16M count=4 iflag=fullblock && \
                sleep 1 && \
                zpool create -f -R /tmp/zfs zroot1 /tmp/zfs.img
      - name: zfs-img
        description: copies out the ZFS image
        steps:
          - copy:
              from: zfs-img-gen
              src: /tmp/zfs.img
              dst: /
```

and in docs mention `make zfs-img`